tests/Storage.test.ts

```diff
@@ -345,4 +346,111 @@ describe('LedgerStorage', () =>
+    {
+        assert.ok(!err1, err1?.message);
+
+        putAllBlockData(ledger_storage, (err2: Error | null) =>
+        {
+            assert.ok(!err2, err2?.message);
+
+            let height: number = 0;
+            ledger_storage.getEnrollments(height,
+                () =>
+                {
+                    it ('Tests that sending a pre-image with a distance of 6 works', (doneIt: () => void) =>
+                    {
+                        let pre_image: PreImageInfo = new PreImageInfo();
+                        pre_image.parseJSON(sample_preImageInfo);
+                        ledger_storage.updatePreImage(pre_image)
+                        .then(() => {
+                            ledger_storage.getValidators(height,
+                                (rows: any[]) =>
+                                {
+                                    assert.strictEqual(rows[0].preimage_distance, 6);
+                                    assert.strictEqual(rows[0].preimage_hash, sample_preImageInfo.hash);
+                                },
+                                (err1: Error) =>
+                                {
+                                    assert.ok(!err1, err1?.message);
+                                    doneIt();
+                                });
+                        })
+                        .catch((err2) => {
+                            assert.ok(!err2, err2);
+                            doneIt();
+                        });
+                    });
+
+                    it ('Fail tests that sending a pre-image with a distance of 5 works', (doneIt: () => void) =>
+                    {
+                        sample_preImageInfo.distance = 5;
+                        let pre_image: PreImageInfo = new PreImageInfo();
+                        pre_image.parseJSON(sample_preImageInfo);
+                        ledger_storage.updatePreImage(pre_image)
+                        .then(() => {
+                            ledger_storage.getValidators(height,
+                                (rows: any[]) =>
+                                {
+                                    assert.strictEqual(rows[0].preimage_distance, 6);
+                                    assert.strictEqual(rows[0].preimage_hash, sample_preImageInfo.hash);
+                                },
+                                (err1: Error) =>
+                                {
+                                    assert.ok(!err1, err1?.message);
+                                    doneIt();
+                                });
+                        })
+                        .catch((err2) => {
+                            assert.ok(!err2, err2);
+                            doneIt();
+                        });
+                    });
+
+                    it ('Preimage distance Greater Than ValidatorCycle Test', (doneIt: () => void) =>
```

Comment:
Name is less than ideal, remember my advice during the previous round of review

Reply:
Thanks for advice.
I change the wording.
it('tests that sending a pre-image with a distance of 1008 works')